### PR TITLE
Implement pickby

### PR DESCRIPTION
**Pull request by svozza**
Implement pickBy as discussed in #195 . I also noticed that we used hasOwnProperties a lot but without accounting for objects created with Object.create(null) so I tidied that up and added tests for it too.

### CODE REVIEW

File: lib/index.js [hunk 1]
```diff
@@ -1556,10 +1549,15 @@ Stream.prototype.pick = function (properties) {
             push(err, x);
         }
         else if (_.isObject(x)) {
+            for (var k in x) {
+                if (hasOwn.call(x, k)) {
+                    properties.push(k);
```
Comment (vqvu) [inline lib/index.js]: Can't you just do the `if(f(key, x[key]))` block here? What's the point of pushing to an array just to immediately iterate over it later?

**Comment by vqvu** [comment]
Hi @svozza.

Thanks for this! And thanks for catching the bad `hasOwnProperties` usages. Can you move the `hasOwnProperties` fixes to a different PR so that they're logically separate? You can then rebase this PR on top of the other once it's merged.

Also, I've been looking at the ramda and lodash implementations, and I don't think you need to `hasOwn` check. Both libraries allow you to pick from all enumerable properties.

For instance
```javascript
require('lodash').pick(Object.create({a: 1}), 'a')
// => {a: 1}

require('ramda').pick('a', Object.create({a: 1}))
// => {a: 1}
```

File: lib/index.js (hunk 1 above)
Comment (svozza) [inline lib/index.js]: You're completely right, dunno what I was thinking there. Fixed.

**Comment by svozza** [comment]
So I removed the hasOwnProps check from pickBy and added a few tests around it too.

**Pull request merged**